File: backend/src/main/java/com/gxs/bank/repository/UpiIdRepository.java
package com.gxs.bank.repository;

import com.gxs.bank.model.UpiId;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

import java.util.List;
import java.util.UUID;

@Repository
public interface UpiIdRepository extends JpaRepository<UpiId, UUID> {
    List<UpiId> findByUserId(UUID userId);
}

File: backend-python/com/gxs/bank/repository/UpiIdRepository.py
from com.gxs.bank.model.UpiId import UpiId
from com.gxs.bank.repository._base import BaseRepository


class UpiIdRepository(BaseRepository):
    def __init__(self, db):
        super().__init__(db, UpiId)

    def findByUserId(self, userId: str):
        return self.db.query(UpiId).filter(UpiId.userId == userId).all()
